hypertrace-graphql-entity-schema/src/main/java/org/hypertrace/graphql/entity/dao/GatewayServiceEntityEdgeFetcher.java

```diff
@@ -119,16 +120,17 @@ private Maybe<Edge> buildEdge(
 
     return zip(
             this.attributeMapConverter.convert(
-                edgeSetGroupRequest.attributeRequests(), response.getAttributeMap()),
+                edgeSetGroupRequest.getAllAttributeRequests(), response.getAttributeMap()),
```

Comment:
this should be the attribute requests originating from this particular edge set - something like `edgeSetGroupRequest.edgeSetRequests().get(neighbor.type()).attributeRequests()`

Reply:
oh neighbour has this entity type set. I checked the entity but the issue was with getting the attributes. changed the implementation